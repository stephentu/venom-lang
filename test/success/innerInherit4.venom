class Outer
  class A 
    def foo() = print("HI FROM A"); end 
  end
  class B
    class C <- A
      def foo() = 
        super.foo(); 
        print("HI FROM C"); 
      end
    end
    def mkC() -> C = return C(); end
  end
  def mkB() -> B = B(); end
end
o = Outer();
b::Outer.B = o.mkB();
c::Outer.B.C = b.mkC();
c.foo();
