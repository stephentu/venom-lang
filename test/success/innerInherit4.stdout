HI FROM A
HI FROM C
